### PR TITLE
[WIP] Add pagination support (ApiOptions overloads) to Watched client.

### LINKED ISSUE #1184: Add ApiOption overloads to methods on IWatchedClient
As part of completing #760 a number of overloads need to be added to the client interfaces.

As an example, this has been added to [`IReleasesClient`](https://github.com/octokit/octokit.net/commit/03ea83148995711fc47148489e2123e65f33bd82)and [`IObservableReleasesClient`](https://github.com/octokit/octokit.net/commit/aa3ce0db9210b778414e290af7f17c4a646bdeec).

The checklist for porting each method is:

- Add an overload for this method on `IWatchedClient` with a new parameter `ApiOptions`
- Implement the new method on the class which implements `IWatchedClient`
- Add an overload for this method on `IObservableWatchedClient` with a new parameter `ApiOptions`
- Implement the new method on the class which implements `IObservableWatchedClient`

This needs to be done for each of these methods:

- [ ] `Task<IReadOnlyList<Repository>> GetAllForCurrent()`
- [ ] `Task<IReadOnlyList<Repository>> GetAllForUser(String user)`
- [ ] `Task<IReadOnlyList<User>> GetAllWatchers(String owner, String name)`

### PR DESCRIPTION
**Pull request by devkhan**
Fixes #1184.

Checklist:

- [x] Add overloads to IWatchedClient + tests.
- [ ] Add overloads to IObservableWatchedClient + tests.

/cc @shiftkey

### CODE REVIEW

File: Octokit.Reactive/Clients/ObservableWatchedClient.cs [hunk 1]
```diff
@@ -52,6 +82,21 @@ public IObservable<Repository> GetAllForUser(string user)
...
+        /// <param name="user">The login of the user</param>
+        /// <param name="options">Options for changing the API's response.</param>
+        /// <exception cref="AuthorizationException">Thrown if the client is not authenticated</exception>
+        /// <returns>A <see cref="IObservable{Repository}"/> watched by the specified user</returns>
+        public IObservable<Repository> GetAllForUser(string user, ApiOptions options)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(user, "user");
+            Ensure.ArgumentNotNull(options, "options");
+
             return _connection.GetAndFlattenAllPages<Repository>(ApiUrls.WatchedByUser(user));
```
Comment (shiftkey) [inline Octokit.Reactive/Clients/ObservableWatchedClient.cs]: Don't forget to pass `options` in here...

Comment (devkhan) [inline Octokit.Reactive/Clients/ObservableWatchedClient.cs]: Oops! Will correct it. :+1:

**Comment by dampir** [comment]
Hi, @devkhan! Are you going to continue your work on this issue in near future :smile:? Just in order to do #1120 pagination support milestone should be done, so if you are not going to continue in near future I'm mark this one to do by myself :smile:

**Comment by devkhan** [comment]
@dampir I know this is getting stale, but I'm really swamped right now between my exams and GSoC. I'll probably complete this after 5th. If this needs to be done before that, go ahead. Otherwise, I'll do it later.

I'm sorry for the delay.

**Comment by dampir** [comment]
@devkhan thank you for fast response, I've marked this issue in my notepad. Anyway there is a lot of other "pagination support" issues that should be done before, so there is chance that I can not make it in time before 5th June :smile:

**Comment by devkhan** [comment]
@dampir Good. I hope you complete all of these fast enough, but if there's too much work, I'd hop in. :+1:

**Comment by devkhan** [comment]
This has been completed in #1326.

**Pull request closed without merging**